doc/Building.md

```diff
@@ -0,0 +1,199 @@
+# Setting up a build environment for KOReader
+
+These instructions are intended to build the emulator in Linux and MacOS. Windows users are suggested to develop in a [Linux VM](https://www.howtogeek.com/howto/11287/how-to-run-ubuntu-in-windows-7-with-vmware-player/) or using the [Windows Subsystem for Linux](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux).
+
+If you only want to work with Lua frontend stuff, you can grab the AppImage and
+run it with `--appimage-extract`.
+
+You can skip most of the following instructions if desired, and use our premade Docker image instead. In that case the only requirements are Git and Docker. See [the virtual development environment README](https://github.com/koreader/virdevenv) for more information.
+
+## Prerequisites
+
+To get and compile the source you must have `patch`, `wget`, `unzip`, `git`,
+`cmake` and `luarocks` installed, as well as a version of `autoconf`
+greater than 2.64. You also need `nasm` and of course a compiler like `gcc`
+or `clang`. If you want to cross-compile for other architectures, you need a proper
+cross-compile toolchain. Your GCC should be at least version 4.8.
+
+### Debian/Ubuntu and derivates
+
+Install the prerequisites using dpkg:
+
+```
+sudo apt-get install build-essential git patch wget unzip \
+gettext autoconf automake cmake libtool nasm luarocks libsdl2-dev \
+libssl-dev libffi-dev libsdl2-dev libc6-dev-i386 xutils-dev linux-libc-dev:i386 zlib1g:i386
+```
+
+
+### Fedora/Red Hat
+
+Install `libstdc++-static`, `SDL` and `SDL-devel` packages using `yum`
```

Comment:
Yum has been replaced by DNF. I don't know the specifics because I don't really use Fedora. I think the commands are essentially the same, just with `dnf` instead of `yum`. Not that we're giving any actual commands here, lol.

```suggestion
Install `libstdc++-static`, `SDL` and `SDL-devel` packages using DNF:

```
sudo dnf install libstdc++-static SDL SDL-devel
```

Not 100 % sure that'll actually work on post-2014/2015 Fedora. Package names might've changed. I'll probably check in a VM (or possibly on bare metal) later.